olmo/config.py

```diff
@@ -831,6 +837,31 @@ class FSDPConfig(BaseConfig):
     """
 
 
+@dataclass
+class SingleGPUConfig(BaseConfig):
+    device: str = "auto"
+    """
+    Device to run single-device training.
+    """
+
+    def get_device(self):
+        if self.device == "auto":
+            if torch.backends.mps.is_available():
+                return torch.device("mps")
+            elif torch.cuda.is_available():
+                return torch.device("cuda")
+            else:
+                return torch.device("cpu")
+        elif self.device == "mps" and not torch.backends.mps.is_available():
+            print("MPS not available.")
+            return torch.device("cpu")
+        elif self.device == "cuda" and not torch.cuda.is_available():
+            print("CUDA not available.")
+            return torch.device("cpu")
```

Comment:
If the device is explicitly set to something that won't work, we should just fail instead of silently fixing it. Throw an exception in this case, like `OLMoConfigurationError`.

We should _never_ just print something. Sometimes we log something (through the logging mechanism). If we need things to fail, we throw exceptions. Also, printing something will go to stdout, where it does not belong in the case of failures.